play-json/jvm/src/main/scala/play/api/libs/json/EnvReads.scala

```diff
@@ -415,6 +419,82 @@ trait EnvReads {
     case _ => JsError(JsonValidationError("error.expected.jsstring"))
   }
 
+  private def javaDurationNumberReads(unit: TemporalUnit, esuffix: String) =
+    Reads[JDuration] {
+      case JsNumber(n) if !n.ulp.isWhole =>
+        JsError(s"error.invalid.$esuffix")
+
+      case JsNumber(n) => JsSuccess(JDuration.of(n.toLong, unit))
```

Comment:
You should probably return an error if the number isn't representable as long. `BigDecimal#toLong` will start providing nonsense values if the number is outside the range of a long, for example
```
scala> val longValue = BigDecimal("1E20").toLong
longValue: Long = 7766279631452241920
```